File: 이것이 취업을 위한 코딩 테스트다/BFS_DFS/DFS/DFS.java
import java.util.ArrayList;

public class DFS {
  public static boolean[] visited= new boolean[9];
  public static ArrayList<ArrayList<Integer>> graph = new ArrayList<ArrayList<Integer>>();

  public static void dfs(int num) {
    System.out.println(num);
    for (int i = 0; i < graph.get(num).size(); i++) {
      visited[num] = true;
      int next = graph.get(num).get(i);

      if (!visited[next]) {
        dfs(next);
      }
    }
  }
  public static void main(String[] args) {
    for (int i =0; i < 9; i++) {
      graph.add(new ArrayList<Integer>());
    }

    graph.get(1).add(2);
    graph.get(1).add(3);
    graph.get(1).add(8);

    graph.get(2).add(1);
    graph.get(2).add(7);

    graph.get(3).add(1);
    graph.get(3).add(4);
    graph.get(3).add(5);

    graph.get(4).add(3);
    graph.get(4).add(5);

    graph.get(5).add(3);
    graph.get(5).add(4);

    graph.get(6).add(7);

    graph.get(7).add(2);
    graph.get(7).add(6);
    graph.get(7).add(8);

    graph.get(8).add(1);
    graph.get(8).add(7);

    dfs(1);
  }
}

File: 이것이 취업을 위한 코딩 테스트다/BFS_DFS/DFS/DFS.cpp
#include <iostream>
#include <vector>
using namespace std;

int visited[9] = {false};
vector<int> graph[9];

void dfs(int num) {
  visited[num] = true;
  cout << num << endl;

  for (int i = 0; i < graph[num].size(); i++) {
    int next = graph[num][i];

    if (!visited[next]) {
      dfs(next);
    }
  }
}

int main() {
  graph[1].push_back(2);
  graph[1].push_back(3);
  graph[1].push_back(8);
  
  graph[2].push_back(1);
  graph[2].push_back(7);

  graph[3].push_back(1);
  graph[3].push_back(4);
  graph[3].push_back(5);

  graph[4].push_back(3);
  graph[4].push_back(5);

  graph[5].push_back(3);
  graph[5].push_back(4);

  graph[6].push_back(7);

  graph[7].push_back(2);
  graph[7].push_back(6);
  graph[7].push_back(8);

  graph[8].push_back(1);
  graph[8].push_back(7);

  dfs(1);
  return 0;
}
